packages/mobile/src/send/SendAmount.tsx

```diff
@@ -189,6 +181,29 @@ function SendAmount(props: Props) {
     secureSendPhoneNumberMapping
   )
 
+  useEffect(() => {
+    if (reviewButtonPressed) {
+      if (recipientVerificationStatus === RecipientVerificationStatus.UNKNOWN) {
+        // Wait until the recipient status is fetched.
+        return
+      } else if (
+        recipientVerificationStatus === RecipientVerificationStatus.UNVERIFIED &&
+        dollarAmount.isGreaterThan(MAX_ESCROW_VALUE)
+      ) {
+        const maxAmount = convertDollarsToLocalAmount(MAX_ESCROW_VALUE, localCurrencyExchangeRate)
+        dispatch(
+          showError(ErrorMessages.MAX_ESCROW_TRANSFER_EXCEEDED, ALERT_BANNER_DURATION, {
+            maxAmount: maxAmount?.toFixed(2),
+            symbol: localCurrencySymbol,
+          })
+        )
+      } else {
+        isOutgoingPaymentRequest ? onRequest() : onSend()
+      }
+      setReviewButtonPressed(false)
+    }
+  }, [reviewButtonPressed, recipientVerificationStatus])
```

Comment:
There's a slight bug here since `localCurrencyExchangeRate` and `localCurrencySymbol` are not the in dependencies list of `useEffect`. Though in practice those values won't change at this point.
Adding them to the dependencies list is not enough though, we'd also need to make sure the action is not dispatched twice. Which would happen if other values change.

Reply:
True, but I feel pretty comfortable leaving it as is. Nothing bad happens if the user is over the limit here; it's an artificial constraint we are placing as an added security measure. There is also no way to change `localCurrency` on the screen.

Is there a use case you had in mind where this could be problematic?

Reply:
Again it's harmless, `localCurrency` doesn't change but `localCurrencyExchangeRate` can change.
The point is that it's not entirely right and introduces subtle bug. Which might or might not bite us in the future ;)